src/twisted/web/http.py

```diff
@@ -2666,6 +2662,35 @@ def writeHeaders(self, version, code, reason, headers):
         headerSequence.append(b"\r\n")
         self.transport.writeSequence(headerSequence)
 
+    def writeHeadersObject(self, version, code, reason, headers):
```

Comment:
`HTTPChannel` is a public API sadly, so we need to add a new method and leave the old one for backwards compatibility.

Reply:
Can we add type annotation to this method definition ?


------

Can we have the existing `writeHeaders` method signature to accept both Iterable and `twisted.web.http_headers.Headers` ?


When a `Headers` instance is pass, we skip the sanitization process.

----------

Is this a high CPU usage ?

Maybe make `twisted.web.http_headers.Headers` an iterator ?

Reply:
I can see what the performance overhead of those two options is like.

Reply:
Oh the iterator version won't work because the issue is the sanitization overhead of readding everything to a `Headers` instance, which we don't want. I can measure isinstance.

Reply:
The slow down is ... hard to measure? So I went with isinstance() version.